### PR TITLE
add PIOc_init_async_from_comms and test

### PR DESCRIPTION
**Pull request by jedwards4b**
Add another interface for PIO_init_async:
int                                                                                                                                                                                          
PIOc_init_async_from_comms(MPI_Comm world, int component_count, MPI_Comm *comp_comm, MPI_Comm io_comm, int rearranger, int *iosysidp)   

this allows the user to call pio_init with the MPI_Comm object already defined for each component and the IO tasks.

### CODE REVIEW

**Review by edwardhartnett — APPROVED** [review]

**Review by edwardhartnett — APPROVED** [review]

File: tests/general/Makefile.am [hunk 1]
```diff
@@ -67,7 +67,7 @@ TESTS = run_tests.sh
 endif # RUN_TESTS
 
 %.F90: %.F90.in
-	util/pio_tf_f90gen.pl --annotate-source --out=$@ $<
+        util/pio_tf_f90gen.pl --annotate-source --out=$@ $<
```
Comment (edwardhartnett) [inline tests/general/Makefile.am:70]: This needs to be a tab char, not spaces.

Comment (jedwards4b) [reply tests/general/Makefile.am]: thanks

**Comment by jedwards4b** [comment]
@edhartnett Do you see what the problem is?  All tests are passing locally.

**Comment by jedwards4b** [comment]
Why was .F90.in in the .gitignore file?

**Comment by edwardhartnett** [comment]
Don't know, it should not have been.

File: src/clib/pioc.c [hunk 2]
```diff
@@ -2085,14 +2085,14 @@ PIOc_init_async_from_comms(MPI_Comm world, int component_count, MPI_Comm *comp_c
         return check_mpi(NULL, NULL, ret, __FILE__, __LINE__);
 
     /* Get num_procs_per_comp for each comp and share with world */
-    num_procs_per_comp = (int *) calloc(component_count, sizeof(int));
+    num_procs_per_comp = (int *) malloc(component_count * sizeof(int));
```
Comment (edwardhartnett) [inline src/clib/pioc.c:2088]: You are changing these so they are no longer pre-filled with zeros. Is that intentional?

Comment (jedwards4b) [reply src/clib/pioc.c:2088]: I am 0 filling them below - calloc seemed to be giving an error in address sanitizer.   Currently the test
pio__iosystem_async_tests passes with mpich and mpt but fails with openmpi.

Comment (edwardhartnett) [reply src/clib/pioc.c:2088]: OK, you want to mention that in the comments somewhere...

Comment (jedwards4b) [reply src/clib/pioc.c:2088]: I'm not really sure that was the problem.  Now that I have it working I could try putting back the callocs.

Comment (edwardhartnett) [reply src/clib/pioc.c:2088]: OK, if you want to just leave it, it's not a big deal...

Generally if I see mallocs + setting to zero I will automatically refactor to callocs whenever I tough the code anyway. ;-)

**Pull request merged**